lib/ide/src/controller/module.rs

```diff
@@ -0,0 +1,64 @@
+//! Module Controller.
+//!
+//! The module controller keeps cached module state (module state is AST+Metadata or equivalent),
+//! and uses it for synchronizing state for text and graph representations. It provides method
+//! for registering text and graph changes. If for example text represntation will be changed, there
+//! will be notifications for both text change and graph change.
+//!
+//! This module is still on WIP state, for now it contains stubs only.
+
+use crate::prelude::*;
+
+use shapely::shared;
+
+
+/// Structure uniquely identifying module location in the project.
+/// Mappable to filesystem path.
+#[derive(Clone,Debug,Eq,Hash,PartialEq)]
+pub struct Location(pub String);
+
+impl Location {
+    /// Obtains path (within a project context) to the file with this module.
+    pub fn to_path(&self) -> file_manager_client::Path {
+        // TODO [mwu] Extremely provisional. When multiple files support is
+        //            added, needs to be fixed, if not earlier.
+        let Location(string) = self;
+        let result = format!("./{}.luna", string);
+        file_manager_client::Path::new(result)
+    }
+}
+
+shared! { Handle
+    /// State data of the module controller.
+    #[derive(Debug)]
+    pub struct State {
+        /// This module's location.
+        location : Location,
+    }
+
+    impl {
+        /// Create new Module controller for given location.
+        pub fn new(location:Location) -> Self {
+            State {location}
+        }
+
+        /// Obtain clone of location.
+        pub fn location_clone(&self) -> Location {
+            self.location.clone()
+        }
+
+        /// Get location as filesystem path.
+        pub fn location_as_path(&self) -> file_manager_client::Path {
+            self.location.to_path()
+        }
```

Comment:
I'm wondering if we should not put the String inside `Location` into `Rc`, so its clone would be super cheap, and then instead of such specialized functions, we can just have an api `my_handle.location().into_path()`